src/openrct2-ui/UiContext.cpp

```diff
@@ -526,6 +526,15 @@ class UiContext final : public IUiContext
 #endif
                 case SDL_KEYDOWN:
                 {
+#if !(defined(__MACOSX__) || defined(__WINDOWS__))
+                    // Ignore winkey keydowns. Handles edge case where *NIX
+                    // tiling window managers don't eat the keypresses when
+                    // changing workspaces.
+                    if (SDL_GetModState() & KMOD_GUI)
+                    {
+                        break;
+                    }
+#endif
```

Comment:
Like I commented, I would move this to the `Platform` file and call this function `shouldIgnoreModKey()`. Then you can clean up the documentation from here and also the `ifdefs`. 

This would then become:
```suggestion
                    if (shouldIgnoreModKey(HSDL_GetModState() & KMOD_GUI))
                    {
                        break;
                    }
```

On the Mac and Windows platform files you would have:
```c++
bool shouldIgnoreModKey(bool hasMod)
{
    return hasMod;
}
```

And on the `Linux` platform file:
```c++
bool shouldIgnoreModKey(bool)
{
    return false;
}
```

This will prevent `ifdef`s from spreading into the code.